### PR TITLE
python unit tests: adds test_clique and adds functions in test_community

### PR DESCRIPTION
**Pull request by PetholzA**
this PR improves #1062

### CODE REVIEW

File: networkit/test/test_clique.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env python3
+import unittest
+import os
+import networkit as nk
+import numpy as np
+import scipy
+
+class Test_Cliques(unittest.TestCase):
```
Comment (angriman) [inline networkit/test/test_clique.py:8]: To be consistent with other Python tests:
```suggestion
class TestCliques(unittest.TestCase):
```

File: networkit/test/test_clique.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+	def test_maxClique(self):
```
Comment (angriman) [inline networkit/test/test_clique.py:10]:
```suggestion
	def testmaxClique(self):
```

File: networkit/test/test_community.py [hunk 3]
```diff
@@ -87,7 +125,7 @@ def testJaccardMeasure(self):
 		PLPLLP = PLPLL.getPartition()
 		JM = nk.community.JaccardMeasure()
 		self.assertIsInstance(JM.getDissimilarity(self.LL, PLMLLP, PLPLLP),float)
-
+
```
Comment (angriman) [inline networkit/test/test_community.py:128]: Drop this change.

File: networkit/test/test_community.py [hunk 4]
```diff
@@ -198,8 +245,16 @@ def testPLP(self):
...
+		self.assertIsInstance(PLPL.getTiming(), list)
 	
-	def testCutClustering(self):
+	def testSpectralPartitioner(self):
+		G = nk.Graph(10)
+		spec = nk.community.SpectralPartitioner(G, 1)
+		spec.run()
+		self.assertEqual(spec.getPartition().numberOfSubsets(), 0)
+		
+	def testCutClustering2(self):
```
Comment (angriman) [inline networkit/test/test_community.py:257]:
```suggestion
	def testCutClustering(self):
```

File: networkit/test/test_community.py [hunk 5]
```diff
@@ -77,7 +116,6 @@ def testHubdominance(self):
 		HD = nk.community.HubDominance()
 		self.assertIsInstance(HD.getQuality(PLMLLP, self.LL),float )
 
-
```
Comment (angriman) [inline networkit/test/test_community.py:80]: Drop this change.

File: networkit/test/test_community.py [hunk 6]
```diff
@@ -218,15 +273,6 @@ def testLFM(self):
...
 			self.assertGreaterEqual(len(cover.subsetsOf(u)), 1)
-	
-	def testPLM(self):
-		G = nk.readGraph("input/PGPgiantcompo.graph",nk.Format.METIS)
-		comms = nk.community.detectCommunities(G, nk.community.PLM(G))
-
-		# each node must be assigned
-		for v in G.iterNodes():
-			self.assertTrue(comms.contains(v))			
-
```
Comment (angriman) [inline networkit/test/test_community.py:276]: Why is this deleted?

File: networkit/test/test_community.py [hunk 7]
```diff
@@ -31,6 +31,45 @@ def testCutClustering(self):
...
+
+	def testIntrapartitionDensity(self):
+		G = nk.community.PLP(self.L).run()
+		Gpart = G.getPartition()
+		dense=nk.community.IntrapartitionDensity(self.L, Gpart)
+		dense.run()
+		self.assertIsInstance(dense.getWeightedAverage(), float)
+		self.assertIsInstance(dense.getUnweightedAverage(), float)
+		self.assertIsInstance(dense.getValue(0), float)
+		self.assertIsInstance(dense.getGlobal(), float)
```
Comment (angriman) [inline networkit/test/test_community.py:60]: We should check that these values are correct, not that they are floats.

File: networkit/test/test_clique.py [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+		G = nk.readGraph("input/jazz2_directed.gml",nk.Format.GML)
+		G.removeSelfLoops()
+		A = nk.clique.MaximalCliques(G, callback=[1,2]).run().getCliques()
```
Comment (angriman) [inline networkit/test/test_clique.py:13]: According to the [documentation](https://networkit.github.io/dev-docs/python_api/clique.html), `callback` should be a callable, but here you are passing a list. Please clarify what are you trying to test here.

**Review by angriman — CHANGES_REQUESTED** [review]

File: networkit/test/test_clique.py [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+	
+	def testMaxClique(self):
+		G = nk.readGraph("input/jazz2_directed.gml",nk.Format.GML)
+		G.removeSelfLoops()
+		#without callback, testing maxOnly, expecting single Clique
+		A1 = nk.clique.MaximalCliques(G, maximumOnly=True).run().getCliques()
+		self.assertEqual(len(A1), 1)
+		#with callback
+		def testCallback():
+			return [1,2,3]
```
Comment (angriman) [inline networkit/test/test_clique.py:19]: I don't think this is how the callback is supposed to be used. Please check the `MaximalClique` C++ tests for an example -- I am not suggesting to test the algorithm twice, just make sure that the callback works correctly also in Python.

File: networkit/test/test_clique.py [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+		A2 = nk.clique.MaximalCliques(G, testCallback()).run().getCliques()
+		self.assertListEqual(A2, [[2]])
```
Comment (angriman) [inline networkit/test/test_clique.py:21]: Move this to a separate `testMaxCliqueWithCallback` test.

File: networkit/test/test_clique.py [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+		G.removeSelfLoops()
+		#without callback, testing maxOnly, expecting single Clique
+		A1 = nk.clique.MaximalCliques(G, maximumOnly=True).run().getCliques()
+		self.assertEqual(len(A1), 1)
+		
+	def testMaxCliqueWithCallback(self):
+		G = nk.readGraph("input/jazz2_directed.gml",nk.Format.GML)
+		G.removeSelfLoops()
+		callback = lambda x,y : 1,2
+		A2 = nk.clique.MaximalCliques(G, callback).run().getCliques()
```
Comment (fabratu) [inline networkit/test/test_clique.py:22]: This is not the correct way to call `MaximalCliques` with a callback function. The second parameter is a bool.

**Comment by fabratu** [comment]
Coverage of the `community` module is at 56%. However, based on an offline discussion, I rather want to finish this PR and add the remaining (or at least get above 80%) tests in a future PR.

File: networkit/test/test_community.py [hunk 12]
```diff
@@ -198,7 +295,18 @@ def testPLP(self):
...
+		self.assertIsInstance(PLPL.getTiming(), list)
 	
+	def testSpectralPartitioner(self):
+		#small instance
+		specSm = nk.community.SpectralPartitioner(self.L, 3)
+		specSm.run()
+		#medium instance
+		jazz = nk.readGraph("input/jazz.graph",nk.Format.METIS)
+		spec = nk.community.SpectralPartitioner(jazz, 4, balanced=False)
+		spec.run()
```
Comment (fabratu) [inline networkit/test/test_community.py:308]: I think we should be more precise when testing this code, since it is mostly Python-code. I would suggest testing for `getPartition()` after `run()`(also inspect the result concerning expectations). If you do this, better als split small and medium instances into two tests.

**Review by fabratu — CHANGES_REQUESTED** [review]

**Review by fabratu — APPROVED** [review]

**Comment by fabratu** [comment]
PR has been stable for some time, the only part missing was the fix in #1108. Merging this early as a result.
